### DISCUSSION TITLE
Random selection

### DISCUSSION THREAD

**Discussion by ryanopily** [UNC-Project-Open-AAC/OS-DPI#105] [Ideas]
Our lab wants to design an OS-DPI layout with custom audio recordings of a selection of phrases. For each phrase, there are a couple of recordings with slight variation between them. The idea is to randomly select one of these variations to play whenever the phrase should be spoken.

To do this, I suppose there should be some sort of random function in the 'eval.js:Functions' object. You can use a field (#audio, maybe) with an array defined in the spreadsheet as input, and it selects a random element from that array. I think we can do this with comma separated values under a field (audio1.mp3, audio2.mp3), and since all field values seem to be interpreted as strings, we would have to split it into an array as input to the random function.

What do you think?

**Comment by gbishop** [comment]
Excellent. Maybe call the function random_choice? It takes a string, splits
it on commas then chooses one of the results randomly.

I could imagine shorthand notations like audio*.mp3 matches any names that
begin with audio and end with .mp3 but comma separated is more explicit.

gb
